### ISSUE TITLE
Error in using tensorflow model with opencv dnn module.

### ISSUE THREAD

**Issue by naserpiltan** [opencv/opencv#17364]
Hi.
Im working on a road segmentation project with tensorflow. I searched and find this usefull repo :
[https://github.com/MaybeShewill-CV/lanenet-lane-detection](https://github.com/MaybeShewill-CV/lanenet-lane-detection)
Author has provided a link to pre-trained checkpoints file as this :
[https://www.dropbox.com/sh/tnsf0lw6psszvy4/AAA81r53jpUI3wLsRW6TiPCya?dl=0](https://www.dropbox.com/sh/tnsf0lw6psszvy4/AAA81r53jpUI3wLsRW6TiPCya?dl=0)

I downloaded this check points and froze them with freeze_lanenet_model.py script in the mnn_project folder of project and now i have a model with this name : lanenet.pb 

freeze_lanenet_model main function is this :

def convert_ckpt_into_pb_file(ckpt_file_path, pb_file_path):

    :param ckpt_file_path:
    :param pb_file_path:
    :return:
    """
    # construct compute graph
    with tf.variable_scope('lanenet'):
        input_tensor = tf.placeholder(dtype=tf.float32, shape=[1, 256, 512, 3], name='input_tensor')

    net = lanenet.LaneNet(phase='test', net_flag='vgg')
    binary_seg_ret, instance_seg_ret = net.inference(input_tensor=input_tensor, name='lanenet_model')

    with tf.variable_scope('lanenet/'):
        binary_seg_ret = tf.cast(binary_seg_ret, dtype=tf.float32)
        binary_seg_ret = tf.squeeze(binary_seg_ret, axis=0, name='final_binary_output')
        instance_seg_ret = tf.squeeze(instance_seg_ret, axis=0, name='final_pixel_embedding_output')

    # create a session
    saver = tf.train.Saver()

    sess_config = tf.ConfigProto()
    sess_config.gpu_options.per_process_gpu_memory_fraction = 0.85
    sess_config.gpu_options.allow_growth = False
    sess_config.gpu_options.allocator_type = 'BFC'

    sess = tf.Session(config=sess_config)

    with sess.as_default():
        saver.restore(sess, ckpt_file_path)

        converted_graph_def = tf.graph_util.convert_variables_to_constants(
            sess,
            input_graph_def=sess.graph.as_graph_def(),
            output_node_names=[
                'lanenet/input_tensor',
                'lanenet/final_binary_output',
                'lanenet/final_pixel_embedding_output'
            ]
        )

        with tf.gfile.GFile(pb_file_path, "wb") as f:
            f.write(converted_graph_def.SerializeToString())`

Now i wanna use this saved model in opencv dnn module like this :
`Net net = readNetFromTensorflow("lanenet.pb");`

But when i run the code , i have this error :
`OpenCV(4.2.0) /home/eagle-soft/opencv/modules/dnn/src/tensorflow/tf_importer.cpp:544: error: (-2:Unspecified error) Input layer not found: lanenet_model/vgg_frontend/vgg16_decode_module/instance_seg_decode/decode_stage_1_fuse/deconv/deconv/Shape in function 'connect'
`
I searched a lot but couldn't find a same issue and solution.
Please give me some workaround to solve this annoying problem.
Thanks in advande.

**Comment by dkurt** [comment]
> OpenCV(4.2.0)

Please try the latest release or even better the latest master branch.

**Comment by naserpiltan** [comment]
@dkurt  thanks for answer
I tried with opencv master branch (opencv 4.3.0) but no thing changed 

> #include <opencv2/opencv.hpp>
#include <opencv2/dnn/dnn.hpp>
#include <iostream>
using namespace std;
using namespace cv;
using namespace cv::dnn;
int main(int argc, char *argv[])
{
  cout<<CV_VERSION <<endl;
  Net net = readNetFromTensorflow("lanenet_frozen.pb");
  return 0;
}

gives me :

> 4.3.0
terminate called after throwing an instance of 'cv::Exception'
  what():  OpenCV(4.3.0-openvino) ../opencv/modules/dnn/src/tensorflow/tf_importer.cpp:545: error: (-2:Unspecified error) Input layer not found: lanenet_model/vgg_frontend/vgg16_decode_module/instance_seg_decode/decode_stage_1_fuse/deconv/deconv/Shape in function 'connect'

**Comment by naserpiltan** [comment]
@alalek  does this problem is related to the opencv version ?

**Comment by LupusSanctus** [comment]
I've obtained the [LaneNet](https://github.com/MaybeShewill-CV/lanenet-lane-detection) in the following way:
freeze_lanenet_model.py -w <PATH_TO_CKPT>/tusimple_lanenet/tusimple_lanenet.ckpt -s <PATH_TO_DIR>/lanenet.pb

The actual config file contains _BiseNetv2_ as the model front end:
 ` FRONT_END: 'bisenetv2'`

[Provided weight files](https://www.dropbox.com/sh/0b6r0ljqi76kyg9/AADedYWO3bnx4PhK1BmbJkJKa?dl=0) in the repository are also for _BiseNetv2_ as model front end. The actual config, checkpoints and `.pb` can be found in the attached archive.

Feeding the frozen model into OpenCV `cv2.dnn.readNetFromTensorflow(...)`:
```
   cvNet = cv.dnn.readNetFromTensorflow(<PATH_TO_DIR>/lanenet.pb)
```

I've got the below error:
```
ERROR: getLayerShapesRecursively OPENCV/DNN: [Eltwise]: getMemoryShapes() throws exception. inputs=2 outputs=0/1 blobs=0
    # eltwise_layer.cpp:220: error: (-215:Assertion failed) inputs[0][j] == inputs[i][j] in function 'getMemoryShapes')
```

The cause of the failure is below _BiseNetv2_ layer:
`LaneNet_1/bisenetv2_frontend/semantic_branch/stage_5/ce_block_3_repeat_1/context_embedding_block/fused_features`

The core of this problem is incorrect processing of the case, when, for example, there are two tensors: `a` (shape: [1,1,1,n] ) and `b`  (shape: [1,<some_val_1>,<some_val_2>,n]  ). [ `tf.math.add`](https://www.tensorflow.org/api_docs/python/tf/math/add) returns the result `c` (shape: [1,<some_val_1>,<some_val_2>,n] ), whereas OpenCV compares `1` vs ` <some_val_1>` and  `1` vs` <some_val_2>` and returns the above error.
